sources/test.py

```diff
@@ -0,0 +1,32 @@
+import random
+
+import numpy as np
+from sklearn.model_selection import train_test_split
+
+from Tools.LogReg import LogReg
+from Tools.ArgParser import ArgParser
+from Tools.DatasetHandler import DatasetHandler
+
+scores = []
+
+
+def test():
+    args = ArgParser()
+    dataset = DatasetHandler(args, parse=True, train=True)
+    for _ in range(10):
+        size = random.randint(20, 90) / 100
+        x_train, x_test, y_train, y_test = train_test_split(

```

Comment:
For evaluate an algorithm with a dataset of this size, i think train_test_split isn't the good method. (where it's efficient with tens of thousands or hundreds of thousands data)
Maybe have a look to k_fold-cross_validation or at least train_validate_test_split variant.